src/Utils/EndpointValidation.ts

```diff
@@ -0,0 +1,64 @@
+import * as Logger from "../Common/Logger";
+
+export function validateEndpoint(endpointToValidate: string | undefined, allowedEndpoints: string[]): boolean {
```

Comment:
Can the `allowedEndpoints` be `ReadyonlyArray<string> | string[]` so we don't have to convert the ReadonlyArrays to a regular array?